pkg/bgpv1/README.md

```diff
@@ -0,0 +1,311 @@
+
+However, there are times where fine-grained control over an instantiated virtual router's configuration needs to take place. 
+
+To accomplish this a Kubernetes annotation is defined which applies to Kubernetes Node resources.
+
+A single annotation is used to specify a set of configuration attributes to apply to a particular virtual router instantiated on a particular host.
+
+The syntax of the annotation is as follows:
+
+```
+	cilium.io/bgp-virtual-router.{asn}="key=value,..."
+```
+
+The `{asn}` portion should be replaced by the virtual router's local ASN you wish to apply these configuration attributes to.
+
+The following sections outline the currently supported attributes.
+
+*note*: *each following section describes the syntax of applying a single attribute, however the annotation's value supports a comma separated lists of attributes and applying multiple attributes in a single annotation is supported.*
+
+*note*: when duplicate `key=value` attributes are defined the last one will be 
+selected.
+
+#### Router ID Attribute
+
+When Cilium is running on an `IPv4` or a dual-stack `IPv4/6` cluster the `BGP Control Plane` will utilize the `IPv4` addressed used by Cilium for external reach ability. 
+
+This will typically be Kubernetes' reported external IP address but can also be configured with a Cilium agent flag.
+
+When running in `IPv6` single stack or when the administrator needs to manually define the instantiated BGP server's router ID a Kubernetes annotation can be placed on the node.
+
+The annotation takes the following syntax:
+
+```
+cilium.io/bgp-virtual-router.{asn}="router-id=127.0.0.1"
+```
+
+The above annotation syntax should replace `{asn}` with the local ASN of the `CiliumBGPVirtualRouter` you are setting the provided router ID for.
+
+When the `BGPControlPlane` evaluates a `CiliumBGPPeeringPolicy` with a `CiliumBGPVirtualRouter` it also searches for an annotation which targets the aforementioned `CiliumBGPVirtualRouter` local ASN. 
+
+If found it will use the provided router ID and not attempt to use the IPv4 address assigned to the node.
+
+#### Local Listening Port
+
+By default the `GoBGP BGPRouterManager` will instantiate each virtual router without a listening port. 
+
+It is possible to deploy a virtual router which creates a local listening port where BGP connections may take place. 
+
+If this is desired the following annotation can be provided
+
+```
+cilium.io/bgp-virtual-router.{asn}="local-port=45450"
+```
+
+
+## Architecture
+
+The `BGP Control Plane` is split into a `Agent-Side Control Plane` and a `Operator-Side` control plane (not yet implemented).
+
+Both control planes are implemented by a `Controller` which follows the `Kubernetes` controller pattern. 
+
+Both control planes primary listen for `CiliumBGPPeeringPolicy` CRDs, long with other Cilium and Kubernetes resources useful for implementing a BGP control plane. 
+
+## Agent-Side Architecture
+
+## Component Diagram
```

Comment:
Please review the level of those titles. “Agent-Side Architecture” should likely be under “Architecture” (level 3), and “Component Diagram” likely under it (level 4)?